shared/src/search/interactive/util.ts

```diff
@@ -31,3 +33,31 @@ export enum FilterTypes {
 }
 
 export const filterTypeKeys: FilterTypes[] = Object.keys(FilterTypes) as FilterTypes[]
+
+export enum NegatedFilters {
+    repo = '-repo',
+    file = '-file',
+    lang = '-lang',
+    repohasfile = '-repohasfile',
+}
+
+/** The list of filters that are able to be negated. */
+export type NegatableFilters = FilterTypes.repo | FilterTypes.file | FilterTypes.repohasfile | FilterTypes.lang
```

Comment:
```suggestion
export type NegatableFilter = FilterTypes.repo | FilterTypes.file | FilterTypes.repohasfile | FilterTypes.lang
```
since it's a single filter that has this type?